### PR TITLE
16847 Removed ledger detail in consent continue out / consent amalgamation out filings

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* bcgov/entity#16847

*Description of changes:*
- app version = ~7.1.4~ 7.1.6
- removed ledger detail ~for non-staff user~ for all users
- updated validation logic
- updated saveFiling()
- updated unit tests

*Also:*
- removed windows event handler  when local filings views are destroyed

*Also:*
- fixed some filing body margins
- fixed ApiFilingIF.continuationOut object

*Also:*
- fixed some calls that didn't need parameters
- fixed some typings
- removed order details display from CCO ledger item
- removed ledger detail section from CCO and CAO sub-components + all supporting code

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-637-xpplthpp.web.app

File: src/views/AgmExtension.vue [hunk 1]
```diff
@@ -297,6 +297,12 @@ export default class AgmExtension extends Mixins(CommonMixin, DateMixin,
     this.updateFilingData('add', FilingCodes.AGM_EXTENSION, undefined, undefined)
   }
 
+  /** Called just before this component is destroyed. */
+  beforeDestroy (): void {
+    // remove event handler
+    window.onbeforeunload = null
+  }
```
Comment (severinbeauvais) [inline src/views/AgmExtension.vue:304]: I found a bug where, if you go to a local filing and change something then cancel out, refreshing or leaving the dashboard would display an "unsaved changes, are you sure you want to leave?" browser alert.

The bug was due to this event handler, which was set (line 266 above) but not cleared. So I added this here and in all other local filings that set the event handler.

(Ask if you want a more thorough explanation. Or Google it.)

File: src/views/AmalgamationOut.vue [hunk 2]
```diff
@@ -216,7 +216,7 @@
                   :certifiedBy.sync="certifiedBy"
                   :class="{ 'invalid-component': !certifyFormValid && showErrors }"
                   :entityDisplay="displayName()"
-                  :message="certifyText(FilingCodes.AMALGANATION_OUT)"
+                  :message="certifyText(FilingCodes.AMALGAMATION_OUT)"
```
Comment (severinbeauvais) [inline src/views/AmalgamationOut.vue:219]: Fixed typo.

File: src/views/ConsentAmalgamationOut.vue [hunk 3]
```diff
@@ -820,7 +831,7 @@ export default class ConsentAmalgamationOut extends Mixins(CommonMixin, DateMixi
 
     const data: any = {
       [FilingTypes.CONSENT_AMALGAMATION_OUT]: {
-        details: `${this.defaultComment}\n${this.detailComment}`,
+        details: this.isRoleStaff ? `${this.defaultComment}\n${this.detailComment}` : null,
```
Comment (severinbeauvais) [inline src/views/ConsentAmalgamationOut.vue:834]: This is where the filing is saved.

The code to fetch the filing (on resume) works fine as-is.

File: src/views/ConsentAmalgamationOut.vue
Comment (severinbeauvais) [inline src/views/ConsentAmalgamationOut.vue:1]: Although we haven't done much work on Consent for Amalgamation Out, I'm keeping this component updated. At the moment it's almost exactly the same as Consent for Continuation Out.

When we get around to formally implementing this feature, we can update this component as needed.

File: tests/unit/ConsentAmalgamationOut.spec.ts [hunk 4]
```diff
@@ -217,6 +217,9 @@ describe('Consent to Amalgamation Out view', () => {
     // verify new Filing ID
     expect(vm.filingId).toBe(456)
 
+    // verify that detail comment from Legal API is not null
+    expect(vm.savedFiling.consentAmalgamationOut.details).toBe('test')
```
Comment (severinbeauvais) [inline tests/unit/ConsentAmalgamationOut.spec.ts:221]: This is to test line 184 and that saving the filing does what's expected.

File: tests/unit/ConsentAmalgamationOut.spec.ts [hunk 5]
```diff
@@ -385,7 +388,6 @@ describe('Consent to Continue Out for general user and IAs only', () => {
     expect(wrapper.findComponent(Certify).exists()).toBe(true)
     expect(wrapper.findComponent(ConfirmDialog).exists()).toBe(true)
     expect(wrapper.findComponent(CourtOrderPoa).exists()).toBe(false) // staff only
-    expect(wrapper.findComponent(DetailComment).exists()).toBe(true)
```
Comment (severinbeauvais) [inline tests/unit/ConsentAmalgamationOut.spec.ts:388]: No DetailComment component when user is not staff (per line 319).

File: src/components/Dashboard/FilingHistoryList/filings/AgmExtension.vue [hunk 6]
```diff
@@ -6,7 +6,7 @@
   >
     <template #body>
       <div v-if="isFilingComplete">
-        <p class="mt-4">
+        <p class="mt-0">
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/filings/AgmExtension.vue:9]: Before:

image

After:

image

File: src/components/Dashboard/FilingHistoryList/filings/ConsentContinuationOut.vue [hunk 7]
```diff
@@ -8,10 +8,11 @@
       <div>
         <p
           v-if="expiry && !isConsentExpired"
-          class="mt-4"
+          class="mt-0"
```
Comment (severinbeauvais) [inline src/components/Dashboard/FilingHistoryList/filings/ConsentContinuationOut.vue:11]: Before:

image

After:

image

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-637-xpplthpp.web.app

File: src/interfaces/api-filing-interface.ts [hunk 8]
```diff
@@ -96,12 +96,10 @@ export interface ApiFilingIF {
...
       courtOrder?: any
-      details: string
-      foreignJurisdiction: {
-        country: string
-        region?: string
-      }
+      country: string
+      details?: string
       legalName: string
+      region: string // may be null
```
Comment (severinbeauvais) [inline src/interfaces/api-filing-interface.ts:102]: This is the filing data we get back from the Legal API. (Unfortunately it doesn't follow the JSON that is used when PUTTING/POSTING a filing.)

Example:
image

**Review by JazzarKarim — APPROVED** [review]
👍

**Review by PaulGarewal — APPROVED** [review]
Was holding off on approval but LGTM 👍

File: src/views/ConsentAmalgamationOut.vue (hunk 3 above)
Comment (severinbeauvais) [reply src/views/ConsentAmalgamationOut.vue]: This is outdated code -- now we don't save the `details` at all.

File: src/components/Dashboard/FilingHistoryList/filings/ConsentContinuationOut.vue (hunk 7 above)
Comment (severinbeauvais) [reply src/components/Dashboard/FilingHistoryList/filings/ConsentContinuationOut.vue:11]: Note that the screenshots above are out of date -- now we don't display the order details (the paragraph "Six Month Consent to Continue Out" and "a detail").

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-637-xpplthpp.web.app

File: tests/unit/ConsentAmalgamationOut.spec.ts (hunk 5 above)
Comment (severinbeauvais) [reply tests/unit/ConsentAmalgamationOut.spec.ts:388]: And now, there's no DetailComment component when user is staff.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-637-xpplthpp.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-637-xpplthpp.web.app

**Pull request merged**